mesheryctl/internal/cli/root/workspaces/list.go

```diff
@@ -0,0 +1,176 @@
+// Copyright 2024 Layer5, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package workspaces
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/eiannone/keyboard"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/config"
+	"github.com/layer5io/meshery/mesheryctl/internal/cli/root/system"
+	"github.com/layer5io/meshery/mesheryctl/pkg/utils"
+	"github.com/layer5io/meshery/server/models"
+	"github.com/pkg/errors"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var listWorkspaceCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List registered workspaces",
+	Long:  `List name of all registered workspaces`,
+	Example: `
+// List all registered workspace
+mesheryctl exp workspace list --orgID [orgId]
+
+// Documentation for workspace can be found at:
+https://docs.layer5.io/cloud/spaces/workspaces/
+
+`,
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		//Check prerequisite
+
+		mctlCfg, err := config.GetMesheryCtl(viper.GetViper())
+		if err != nil {
+			return utils.ErrLoadConfig(err)
+		}
+		err = utils.IsServerRunning(mctlCfg.GetBaseMesheryURL())
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+		ctx, err := mctlCfg.GetCurrentContext()
+		if err != nil {
+			utils.Log.Error(system.ErrGetCurrentContext(err))
+			return err
+		}
+		err = ctx.ValidateVersion()
+		if err != nil {
+			utils.Log.Error(err)
+			return err
+		}
+		return nil
+	},
+
+	Args: func(cmd *cobra.Command, args []string) error {
+		const errMsg = "Usage: mesheryctl exp workspace create --orgId [orgId] --name [workspace-name] --description [workspace-description]\nRun 'mesheryctl exp workspace create --help' to see detailed help message"
```

Comment:
Update the usages example, the command is for `list` and the mesage showcases `create`.

Reply:
https://github.com/meshery/meshery/pull/10634#discussion_r1550380485

Reply:
https://github.com/meshery/meshery/pull/10634#discussion_r1552242031